widgets/matplotlib/measurement/energy_spectrum.py

```diff
@@ -31,29 +31,24 @@
              "Juhani Sundell"
 __version__ = "2.0"
 
-import copy
```

Comment:
I am not sure if I understood this right. But it looks like it is used here only. So?
 
https://github.com/JYU-IBA/potku/blob/f1998afdca2d1793c45b0646f83fb753d5978a17/widgets/matplotlib/measurement/energy_spectrum.py#L88-L89

https://github.com/JYU-IBA/potku/blob/f1998afdca2d1793c45b0646f83fb753d5978a17/widgets/matplotlib/measurement/energy_spectrum.py#L411-L413

https://github.com/JYU-IBA/potku/blob/f1998afdca2d1793c45b0646f83fb753d5978a17/widgets/matplotlib/measurement/energy_spectrum.py#L530-L531

Reply:
Sharing a mutable value (mostly a list or a dict) between classes is a bit iffy because it's unexpected and can cause issues if modified. Illustration of the issue:

```
class A:
    def __init__(list_):
        self.list = list_
        # ...

class B:
    def __init__(list_):
        self.list = list_
        # ...

a = A([1, 2, 3])
b = B(a.list)

a.list.append(4)
# both a.list and b.list are modified
```

In our case, A is `EnergySpectrumWidget` and B is `MatplotlibEnergySpectrumWidget`. (The shared value is different too.)

To be safe, we could do a `copy.deepcopy()` for `self.simulation_energy` and `self.measurement_energy` before `self.matplotlib = MatplotlibEnergySpectrumWidget(...` in `EnergySpectrumWidget.__init__` (line ~572).

Reply:
Thanks for the clear explanation! I will do that.